zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledCall.java

```diff
@@ -59,38 +69,38 @@
    * anyway.
    */
   @Override protected Void doExecute() throws IOException {
-    AwaitableCallback awaitableCallback = new AwaitableCallback();
-    doEnqueue(awaitableCallback);
-    if (!await(awaitableCallback.countDown)) throw new InterruptedIOException();
-
-    Throwable t = awaitableCallback.throwable;
-    if (t != null) {
-      if (t instanceof Error) throw (Error) t;
-      if (t instanceof IOException) throw (IOException) t;
-      if (t instanceof RuntimeException) throw (RuntimeException) t;
-      throw new RuntimeException(t);
-    }
-    return null; // Void
+    // Enqueue the call invocation on the executor and block until it completes.
+    doEnqueue(NOOP_CALLBACK);
```

Comment:
This is another thing that can affect the GC metrics: constant callback instead of a new one.

I'm thinking the real gains you got from this change is less related to CountDownLatch and more to some of the tertiary changes.